src/Donations/Migrations/RecalculateExchangeRate.php

```diff
@@ -0,0 +1,180 @@
+                ->select('posts.ID')
+                ->orderBy('posts.ID', 'ASC');
+
+            // Migration Runner will pass null for lastId in the last step
+            if (is_null($lastId)) {
+                $query->where('posts.ID', $firstId, '>');
+            } else {
+                $query->whereBetween('posts.ID', $firstId, $lastId);
+            }
+
+            $donations = $query->getAll();
+
+            foreach ($donations as $donation) {
+                try {
+                    $amount = Money::fromDecimal($donation->amount, $donation->currency);
+                    $baseAmount = Money::fromDecimal($donation->baseAmount, $donation->baseCurrency);
+
+                    if ($baseAmount->formatToDecimal() === '0') {
+                        throw new DatabaseMigrationException(
+                            sprintf(
+                                'Invalid base amount (0) for donation ID %d',
+                                $donation->ID
+                            )
+                        );
+                    }
+
+                    /** @var Money $exchangeRate */
+                    $exchangeRate = $amount->divide($baseAmount->formatToDecimal());
+
+                    give()->payment_meta->update_meta($donation->ID, DonationMetaKeys::EXCHANGE_RATE, $exchangeRate->formatToDecimal());
+                } catch (\Exception $e) {
+                    throw new DatabaseMigrationException(
+                        sprintf(
+                            'Failed to process donation ID %d: %s',
+                            $donation->ID,
+                            $e->getMessage()
+                        ),
+                        0,
+                        $e
+                    );
+                }
+            }
+        } catch (DatabaseQueryException $exception) {
+            throw new DatabaseMigrationException(
+                'An error occurred while updating donation exchange rates',
+                0,
+                $exception
+            );
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getItemsCount(): int
+    {
+        return $this->query()->count();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBatchItemsAfter($lastId): ?array
+    {
+        $item = clone $this->query()
+            ->select('MIN(posts.ID) AS first_id, MAX(posts.ID) AS last_id')
+            ->where('posts.ID', $lastId, '>')
+            ->orderBy('posts.ID', 'ASC')
+            ->limit($this->getBatchSize())
+            ->get();
+
+        if (!$item) {
+            return null;
+        }
+
+        return [
+            $item->first_id,
+            $item->last_id,
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getBatchSize(): int
+    {
+        return 100;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function hasMoreItemsToBatch($lastProcessedId): ?bool
+    {
+        return $this->query()
```

Comment:
This is not very performant because we are using attachMeta in the query, and we don't need that here.

Reply:
`attachMeta` is now only present in the `runBatch` method.

1769236f0821a9a640d8de59e54bf0ca7db330fd